Exercise_1.java

```diff
@@ -2,32 +2,57 @@
     static final int MAX = 1000; 
     int top; 
     int a[] = new int[MAX]; // Maximum size of Stack 
-  
-    boolean isEmpty() 
-    { 
-        //Write your code here 
-    } 
 
     Stack() 
     { 
         //Initialize your constructor 
+        top = -1;
+    } 
+  
+    boolean isEmpty() 
+    { 
+        //Write your code here 
+        return top == -1;
     } 
   
     boolean push(int x) 
     { 
         //Check for stack Overflow
         //Write your code here
+        // check if the stack reached max size
+        if(top == MAX-1){
+            return false; // stack overflow
+        }
+        else {
+            top++;
+            a[top] = x;
+            return true;
+        }
     } 
   
     int pop() 
     { 
         //If empty return 0 and print " Stack Underflow"
         //Write your code here
+        if(top==-1){
+            System.out.println("Stack Underflow");
+            return 0;
+        }
+        else {
+            top--;
+            return a[top+1];
+        }
     } 
   
     int peek() 

```

Comment:
when there's only one element, peek returns **ArrayIndexOutOfBoundsException**  Please go through your logic and rectify the error, If you find it difficult, let me know

Reply:
Got it, I should be doing
` a[top--];`